Block MobileGestalt Mach sandbox extension

pvollan opened the pull request:
#### f46a5d13b993ad444ac5802fca3d8a083287f3ca
<pre>
Block MobileGestalt Mach sandbox extension
<a href="https://bugs.webkit.org/show_bug.cgi?id=262116">https://bugs.webkit.org/show_bug.cgi?id=262116</a>
<a href="https://rdar.apple.com/problem/116054248">rdar://problem/116054248</a>

Reviewed by NOBODY (OOPS!).

Block MobileGestalt Mach sandbox extension in the WebContent process on iOS.

* Source/WebCore/PAL/pal/spi/ios/MobileGestaltSPI.h:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::initializeGPUProcess):
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.cpp:
(WebKit::GPUProcessCreationParameters::encode const):
(WebKit::GPUProcessCreationParameters::decode):
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.h:
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.serialization.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Shared/Cocoa/MobileGestaltParameters.cpp: Added.
(WebKit::MobileGestaltParameters::initialize):
(WebKit::MobileGestaltParameters::populate):
* Source/WebKit/Shared/Cocoa/MobileGestaltParameters.h: Added.
* Source/WebKit/Shared/Cocoa/MobileGestaltParameters.serialization.in: Added.
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/GPUProcessProxyCocoa.mm:
(WebKit::GPUProcessProxy::platformInitializeGPUProcessParameters):
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f46a5d13b993ad444ac5802fca3d8a083287f3ca

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25571 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4176 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26856 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27674 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23434 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25854 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5957 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1611 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23583 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25820 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3129 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22043 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28256 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2765 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22994 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29089 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23366 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23359 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26928 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2741 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/992 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4125 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3200 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3078 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/Shared/Cocoa/MobileGestaltParameters.h:
```diff
@@ -0,0 +1,54 @@
+    bool deviceSupportsARKit;
+    bool timeSyncCapability;
+    bool wAPICapability;
+    int mainDisplayRotation;
```
Nit: We could save a bit of memory if you grouped the `bool` values together.

brentfulgham commented on Source/WebCore/PAL/pal/spi/ios/MobileGestaltSPI.h:
```diff
@@ -95,6 +95,8 @@ SInt32 MGGetSInt32Answer(CFStringRef question, SInt32 defaultValue);
 Float32 MGGetFloat32Answer(CFStringRef question, Float32 defaultValue);
 #endif
 
+int MGSetAnswer(CFStringRef question, CFTypeRef answer);
```
Do we need to guard for any particular OS versions? I suspect this is present on all supported iOS targets, but wanted to double-check.

brentfulgham commented on Source/WebKit/Shared/Cocoa/MobileGestaltParameters.cpp:
```diff
@@ -0,0 +1,94 @@
+    hasExtendedColorDisplay = MGGetBoolAnswer(kMGQHasExtendedColorDisplay);
+    deviceCornerRadius = MGGetFloat32Answer(kMGQDeviceCornerRadius, 0);
+    supportsForceTouch = MGGetBoolAnswer(kMGQSupportsForceTouch);
+    bluetoothCapability = MGGetBoolAnswer(kMGQBluetoothCapability);
```
It would be good if we could get Bluetooth knowledge out of the WCP. Seems unlikely there are any cases where the WebContent process should seek this information. Seems like the UIProcess and GPUProcess would be the right places to need this.

But perhaps MobileGestalt fails in some way if this isn't populated accurately.

brentfulgham reviewed (approved):
Looks good! I do not think the test failure is related, but please take a look and land if you believe it is safe.

pvollan committed:
Block MobileGestalt Mach sandbox extension
https://bugs.webkit.org/show_bug.cgi?id=262116
rdar://problem/116054248

Reviewed by NOBODY (OOPS!).

Block MobileGestalt Mach sandbox extension in the WebContent process on iOS.

* Source/WebCore/PAL/pal/spi/ios/MobileGestaltSPI.h:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::initializeGPUProcess):
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.cpp:
(WebKit::GPUProcessCreationParameters::encode const):
(WebKit::GPUProcessCreationParameters::decode):
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.h:
* Source/WebKit/GPUProcess/GPUProcessCreationParameters.serialization.in:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Shared/Cocoa/MobileGestaltParameters.cpp: Added.
(WebKit::MobileGestaltParameters::initialize):
(WebKit::MobileGestaltParameters::populate):
* Source/WebKit/Shared/Cocoa/MobileGestaltParameters.h: Added.
* Source/WebKit/Shared/Cocoa/MobileGestaltParameters.serialization.in: Added.
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Cocoa/GPUProcessProxyCocoa.mm:
(WebKit::GPUProcessProxy::platformInitializeGPUProcessParameters):
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
